.github/actions/fetch_ctk/action.yml

```diff
@@ -1,65 +1,63 @@
-name: Common setup
+name: Fetch mini CTK
+
+description: Fetch (or create) a mini CUDA Toolkit from cache
 
 inputs:
-  client-repo:
-    required: true
-    type: string
-  build-type:
-    required: true
-    type: string
-  target-device:
-    required: true
-    type: string
   host-platform:
     required: true
     type: string
-  build-mode:
-    required: true
-    type: string
-  upload-enabled:
-    required: true
-    type: boolean
-  python-version:
-    required: true
-    type: string
   cuda-version:
     required: true
     type: string
 
 runs:
   using: composite
   steps:
-    # WAR: setup-python is not relocatable...
-    # see https://github.com/actions/setup-python/issues/871
-    - name: Set up Python ${{ inputs.python-version }}
-      if: ${{ startsWith(inputs.host-platform, 'linux') }}
-      id: setup-python
-      uses: actions/setup-python@v5
-      with:
-        python-version: "3.12"
-
-    - name: Set up MSVC
-      if: ${{ startsWith(inputs.host-platform, 'win') }}
-      uses: ilammy/msvc-dev-cmd@v1
-
-    - name: Dump environment
-      shell: bash --noprofile --norc -xeuo pipefail {0}
-      run: |
-        env
-
     - name: Set up CTK cache variable
       shell: bash --noprofile --norc -xeuo pipefail {0}
       run: |
         echo "CTK_CACHE_KEY=mini-ctk-${{ inputs.cuda-version }}-${{ inputs.host-platform }}" >> $GITHUB_ENV
         echo "CTK_CACHE_FILENAME=mini-ctk-${{ inputs.cuda-version }}-${{ inputs.host-platform }}.tar.gz" >> $GITHUB_ENV
 
+    - name: Install dependencies
```

Comment:
This step is really messy, and it is so because we could run this action in different environments (GH- or self- hosted vm images, or an arbitrary container). It'd be really nice if we could unify the environments...